src/lib/components/SelectField/SelectField.jsx

```diff
@@ -69,16 +70,28 @@ export const SelectField = ({
             value={value}
           >
             {
-              options.map((option) => (
-                <option
-                  disabled={option.disabled}
-                  id={id && `${id}__item__${option.value}`}
-                  key={option.value}
-                  value={option.value}
-                >
-                  {option.label}
-                </option>
-              ))
+              options.map((option) => {
+                if ('options' in option) {
+                  return (
+                    <optgroup label={option.label} key={option.label}>
+                      {option.options.map((optgroupOption) => (
+                        <Option
+                          key={optgroupOption.label}
+                          {...optgroupOption}
+                          {...(id && { id: `${id}__item__${optgroupOption.value}` })}
+                        />
+                      ))}
+                    </optgroup>
+                  );
+                }
+                return (
+                  <Option
+                    key={option.label}
```

Comment:
Whenever possible, use `option.value` instead of `option.label`, because `value` is required to be unique. Label should be also unique to make sense, but it is not strong requirement. `value` is safer.

I am 100% OK with key={option.label} which is used for optgroup, but it's better that installing some dependency like nanoid. Decision should be made by @mbohal, who has already commented this few lines above.

Reply:
> value is required to be unique

I'm not sure that is correct. I could not find any mention of it.

Honestly I have no strong opinion about which one is correct - I think they are both equally wrong. Either one will work in 99% of cases. So I agree we should use `value` for `<option>` and `label` for `<optgroup>`.

To deal with the 1% of cases we could allow for passing in the `key` manually by adding a new attribute to the existing `label`, `value`, `disabled` in the option definition object.

Reply:
> Should I also allow passing in the key for <optgroup>

Yes please. And I would add it to he `Radio` ccomponent as well. To have it uniform.

Reply:
Can I include it into the single commit or should I create separate commit for `Radio` component update?

Reply:
> Can I include it into the single commit or should I create separate commit

I think a dedicated commit would be better